src/diffusers/pipelines/faster_cache_utils.py

```diff
@@ -0,0 +1,305 @@
+        timestep_skip_range = config.temporal_attention_timestep_skip_range
+        block_type = "temporal"
+
+    if block_skip_range is None or timestep_skip_range is None:
+        logger.info(
+            f'Unable to apply FasterCache to the selected layer: "{name}" because it does '
+            f"not match any of the required criteria for spatial, temporal or cross attention layers. Note, "
+            f"however, that this layer may still be valid for applying PAB. Please specify the correct "
+            f"block identifiers in the configuration or use the specialized `apply_fastercache_on_module` "
+            f"function to apply FasterCache to this layer."
+        )
+        return
+
+    def skip_callback(module: nn.Module) -> bool:
+        is_using_classifier_free_guidance = pipeline.do_classifier_free_guidance
+        if not is_using_classifier_free_guidance:
+            return False
+
+        fastercache_state = module._fastercache_state
+        is_within_timestep_range = timestep_skip_range[0] < pipeline._current_timestep < timestep_skip_range[1]
+
+        if not is_within_timestep_range:
+            # We are still not in the phase of inference where skipping attention is possible without minimal quality
+            # loss, as described in the paper. So, the attention computation cannot be skipped
+            return False
+
+        should_compute_attention = (
+            fastercache_state.iteration > 0 and fastercache_state.iteration % block_skip_range == 0
+        )
+        return not should_compute_attention
+
+    logger.debug(f"Enabling FasterCache ({block_type}) for layer: {name}")
+    apply_fastercache_on_module(module, skip_callback, config.attention_weight_callback)
+
+
+class FasterCacheModelHook(ModelHook):
+    def __init__(self) -> None:
+        super().__init__()
+
+
+class FasterCacheBlockHook(ModelHook):
+    def __init__(
+        self, skip_callback: Callable[[nn.Module], bool], weight_callback: Callable[[nn.Module], float]
+    ) -> None:
+        super().__init__()
+
+        self.skip_callback = skip_callback
+        self.weight_callback = weight_callback
+
+    def new_forward(self, module: nn.Module, *args, **kwargs) -> Any:
+        args, kwargs = module._diffusers_hook.pre_forward(module, *args, **kwargs)
+
+        if self.skip_callback(module):
+            t_2_output, t_output = module._fastercache_state.cache
+            output = t_output + (t_output - t_2_output) * self.weight_callback(module)
+        else:
+            output = module._old_forward(*args, **kwargs)
+
+        return module._diffusers_hook.post_forward(module, output)
+
+    def post_forward(self, module: nn.Module, output: Any) -> Any:
+        module._fastercache_state.update_state(output)
+        return output
+
+
+# Reference: https://github.com/Vchitect/FasterCache/blob/fab32c15014636dc854948319c0a9a8d92c7acb4/scripts/latte/fastercache_sample_latte.py#L127C1-L143C39
+@torch.no_grad()
+def _fft(tensor):
```

Comment:
Could go to `torch_utils.py` similar to:
https://github.com/huggingface/diffusers/blob/825979ddc3d03462287f1f5439e89ccac8cc71e9/src/diffusers/utils/torch_utils.py#L93